src/datasource.ts

```diff
@@ -296,22 +303,34 @@ export default class CogniteDatasource extends DataSourceApi<
     );
   }
 
-  async fetchEvents(expr: string, timeRange: EventsFilterTimeParams) {
-    const { filters, params } = parseQuery(expr);
+  async fetchEvents({ expr, eventQuery }, timeRange: EventsFilterTimeParams) {
+    let filter = [];
+    let params = {};
+    if (expr) {
+      const parsedQuery = parseQuery(expr);
+      filter = parsedQuery.filters;
+      params = parsedQuery.params;
+    }
+    const advancedFilter =
+      this.connector.isEventsAdvancedFilteringEnabled() && eventQuery.length
+        ? JSON.parse(eventQuery)
+        : undefined;
     const data: FilterRequest<EventsFilterRequestParams> = {
+      advancedFilter,
       filter: { ...timeRange, ...params },
       limit: EVENTS_PAGE_LIMIT,
     };
-
     const items = await this.connector.fetchItems<CogniteEvent>({
       data,
       path: `/events/list`,
       method: HttpMethod.POST,
+      headers: {
+        'cdf-version': 'alpha',
+      },
```

Comment:
this header should only appear if "advancedFilter" is used. otherwise it's unsafe to use in production